### ISSUE TITLE
Support pagination

### ISSUE THREAD

**Issue by clickingbuttons** [polygon-io/client-js#103]
Add `list` methods that paginate through `next_url`.

**Comment by jdconley** [comment]
Yes, it would be handy. I'm currently doing it like this on the 6.x client, when paging over all tickers, for example: 

```
async getAllTickers(): Promise<ITickersResults[]> {
    const results: ITickersResults[] = [];
    let tickers = await this.client.reference.tickers(
      { limit: 1000 },
     { ... headers go here ... },
    );
    results.push(...tickers.results);

    while (tickers.next_url) {
      const url = new URL(tickers.next_url);
      tickers = await this.client.reference.tickers(
        { cursor: url.searchParams.get('cursor') },
        headers,
      );
      if (tickers.results?.length) {
        results.push(...tickers.results);
      } else {
        throw new Error('No results for ticker request');
      }
    }
    return results;
  }
```

I'll abstract it into a little helper the next time I need to page over something, but still would be handy to be builtin perhaps with a generator and a yield for each response rather than building and returning the full list like I did in this simple example.

**Comment by iukea1** [comment]
Please add

**Comment by Mrfoofums** [comment]
Another +1 here. Im about 30 minutes into scaffolding out an API and without pagination the client is kind of pointless. Easier to just stick to axios

**Comment by justinpolygon** [comment]
Hey @Mrfoofums, we have a patch that we're testing to support pagination via https://github.com/polygon-io/client-js/pull/164 hopefully we'll roll this out this coming week. I'll ping you when we do.

**Comment by Mrfoofums** [comment]
Awesome! Can't wait to use it

**Comment by eth0real** [comment]
Very looking forward to this as well!

**Comment by emorling** [comment]
+1

**Comment by RandyBooth** [comment]
+1

**Comment by justinpolygon** [comment]
Hello all,

We have merged and released pagination support in v7.3.0. To make sure we don't break any existing users of the client-js library you can turn this on via a fetch option (example below). We have also added debugging in that you can see what url, client headers, and server headers are being sent. This should help everyone debug what is happening behind the scenes.

Here's the docs:
 - Pagination: https://github.com/polygon-io/client-js#pagination
 - Tracing: https://github.com/polygon-io/client-js#debugging

Here's an example (just add your API key):

```javascript
import('@polygon.io/client-js').then(({ restClient }) => {
	const globalFetchOptions = {
		pagination: true,
		trace: true,
	};
	const rest = restClient("XXXX-APIKEY-HERE-XXXX", "https://api.polygon.io", globalFetchOptions);
	rest.stocks.aggregates("TSLA", 1, "minute", "2022-01-01", "2023-08-31", { limit: 50000 }).then((data) => {
	    const resultCount = data.length;
	    console.log("Result count:", resultCount);
	}).catch(e => {
		console.error('An error happened:', e);
	});
});
```

Hopefully these features come in handy and remove a bunch of complexity.

Cheers,
Team Polygon
